src/keyboard.js

```diff
@@ -176,15 +176,31 @@ Keyboard.prototype.inputLocalOptions = function() {
 };
 
 Keyboard.prototype.printChar = function(char) {
-  var current_val = this.$current_input.val();
-  this.$current_input.val(current_val + char);
-  this.$current_input.focus().trigger("input");
+  var selStart = this.$current_input[0].selectionStart;
+  var selEnd = this.$current_input[0].selectionEnd;
+  var textAreaStr = this.$current_input.val();
+  var value = textAreaStr.substring(0, selStart) + char + textAreaStr.substring(selEnd);
+
+  this.$current_input.val(value).focus();
+  this.$current_input[0].selectionStart = selStart+1, this.$current_input[0].selectionEnd = selStart+1;
+
 };
 
 Keyboard.prototype.deleteChar = function() {
-  var current_val = this.$current_input.val();
-  this.$current_input.val(current_val.slice(0,-1));
-  this.$current_input.focus().trigger("input");
+  var selStart = this.$current_input[0].selectionStart;
+  var selEnd = this.$current_input[0].selectionEnd;
+
+  var textAreaStr = this.$current_input.val();
+  var after = textAreaStr.substring(0, selStart-1);
+  console.log(after);
+
+  var value = after + textAreaStr.substring(selEnd);
+
+  //var value = textAreaStr.slice(selStart, -1);
```

Comment:
@worgock thank you a lot for the attention to this project. I'm really sorry that now i don't have enough time to commit to this project. I added you as contributor to the project and I think it would be fair to allow you commit without my special attention.

Only one comment regarding this PR is that maybe you can clean-up a code little bit and remove commented out code if it's not needed as also `console.log` debugging. 

And thank you one more time :+1: